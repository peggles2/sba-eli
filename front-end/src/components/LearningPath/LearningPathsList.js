import React from "react";
import { Card } from "semantic-ui-react";
import { connect } from "react-redux";
import LearningPathsItem from "./LearningPathsItem";

import { getLearningPaths } from "../../actions/learningPathActions";

export class LearningPathsList extends React.Component {
  componentDidMount() {
    this.props.dispatch(getLearningPaths());
  }

  render() {
    let result = <h3>There are no journeys available to view at this time</h3>;
    if (this.props.learningPaths.length > 0) {
      result = this.props.learningPaths.map(c => (
        <LearningPathsItem
          key={c.id}
          id={c.id}
          name={c.name}
          course_code={c.course_code}
        />
      ));
    }

    return <Card.Group itemsPerRow={3}>{result}</Card.Group>;
  }
}
const mapStateToProps = store => {
  return {
    learningPaths: store.learningPath.learningPaths
  };
};

export default connect(mapStateToProps)(LearningPathsList);
