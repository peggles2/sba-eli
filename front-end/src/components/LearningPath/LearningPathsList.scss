#learningPathCards {
  padding-left: 10%;
}