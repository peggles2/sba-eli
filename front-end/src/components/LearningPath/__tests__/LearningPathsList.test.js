import React from 'react';
import { List } from 'semantic-ui-react';
import LearningPathsList from '../LearningPathsList';
import { shallow, mount } from 'enzyme';

describe('LearningPathsList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<LearningPathsList/>);

    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render a <div>', () => {
    const wrapper = shallow(<LearningPathsList/>);

    expect(wrapper.find('div').length).toEqual(1);
  });

  describe('rendering the list', () => {
    var wrapper;
    
    beforeEach(function() {
      const lps = [
        { id: 1, name: "Course 1" },
        { id: 2, name: "Course 2" },
        { id: 3, name: "Course 3" },
        { id: 4, name: "Course 4" },
      ]

      wrapper = mount(<LearningPathsList/>);
      wrapper.setState({learningPaths: lps})
    })  

    it('should render a <List>', () => {
      expect(wrapper.find(List).length).toEqual(1);
    });
    
    it('should render a 4 <li>', () => {
      expect(wrapper.find(List.Item).length).toEqual(4);
    });
  })
});
