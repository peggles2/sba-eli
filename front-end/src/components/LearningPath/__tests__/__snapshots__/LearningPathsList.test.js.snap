// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningPathsList should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningPathsList
    dispatch={[Function]}
    learningPaths={
      Array [
        Object {
          "id": 1,
          "name": "Course 1",
        },
        Object {
          "id": 2,
          "name": "Course 2",
        },
        Object {
          "id": 3,
          "name": "Course 3",
        },
        Object {
          "id": 4,
          "name": "Course 4",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CardGroup>
        <Connect(withRouter(LearningPathsItem))
          id={1}
          name="Course 1"
        />
        <Connect(withRouter(LearningPathsItem))
          id={2}
          name="Course 2"
        />
        <Connect(withRouter(LearningPathsItem))
          id={3}
          name="Course 3"
        />
        <Connect(withRouter(LearningPathsItem))
          id={4}
          name="Course 4"
        />
      </CardGroup>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(withRouter(LearningPathsItem))
            id={1}
            name="Course 1"
          />,
          <Connect(withRouter(LearningPathsItem))
            id={2}
            name="Course 2"
          />,
          <Connect(withRouter(LearningPathsItem))
            id={3}
            name="Course 3"
          />,
          <Connect(withRouter(LearningPathsItem))
            id={4}
            name="Course 4"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "course_code": undefined,
            "id": 1,
            "name": "Course 1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "course_code": undefined,
            "id": 2,
            "name": "Course 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "course_code": undefined,
            "id": 3,
            "name": "Course 3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "course_code": undefined,
            "id": 4,
            "name": "Course 4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CardGroup>
          <Connect(withRouter(LearningPathsItem))
            id={1}
            name="Course 1"
          />
          <Connect(withRouter(LearningPathsItem))
            id={2}
            name="Course 2"
          />
          <Connect(withRouter(LearningPathsItem))
            id={3}
            name="Course 3"
          />
          <Connect(withRouter(LearningPathsItem))
            id={4}
            name="Course 4"
          />
        </CardGroup>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Connect(withRouter(LearningPathsItem))
              id={1}
              name="Course 1"
            />,
            <Connect(withRouter(LearningPathsItem))
              id={2}
              name="Course 2"
            />,
            <Connect(withRouter(LearningPathsItem))
              id={3}
              name="Course 3"
            />,
            <Connect(withRouter(LearningPathsItem))
              id={4}
              name="Course 4"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "course_code": undefined,
              "id": 1,
              "name": "Course 1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "course_code": undefined,
              "id": 2,
              "name": "Course 2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "course_code": undefined,
              "id": 3,
              "name": "Course 3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "class",
            "props": Object {
              "course_code": undefined,
              "id": 4,
              "name": "Course 4",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
